### PR TITLE
Travis CI: add staticcheck target

### LINKED ISSUE #173: staticcheck / go module issues with golang/circleci:1.11
https://circleci.com/gh/prometheus/statsd_exporter/347#artifacts/containers/0

I can reproduce this locally (on macOS, using https://github.com/dziemba/mobymac) with the golang/circleci:1.11 image but not the golang:1.11.4 image or on Darwin directly.

For now, I'm cutting out staticcheck from the build to unblock the 0.9.0 release candidate.

### PR DESCRIPTION
**Pull request by simonpasquier**
Attempt to close #173

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,5 +9,5 @@ go:
 go_import_path: github.com/prometheus/statsd_exporter
 
 script:
-- make
+- make precheck style unused test staticcheck build
```
Comment (matthiasr) [inline .travis.yml]: or equivalently

```suggestion
- make all
```

**Comment by matthiasr** [comment]
It still doesn't work on CircleCI (see #176), I'm really curious what the difference is between the two?

**Comment by simonpasquier** [comment]
So having `test` before `staticcheck` makes Travis CI happy. Honestly I have no clue why Travis fails to run `staticcheck` first... I'm not seeing the issue on my local machine neither Circle CI but @discordianfish also reported the problem on IRC.

**Comment by matthiasr** [comment]
I originally encountered this on CircleCI.

a dependency declaration like `foo: one two three` does not actually guarantee ordering – in practice, make usually walks the list left-to-right, but it does not know that two should come before three. If this is what we want to force, add a dependency `staticcheck: test` – that way make knows that it needs to test before staticchecking.

if it's not only a problem for this repo, should we make this change in Makefile.common? I'm happy to do it locally first though.

**Comment by simonpasquier** [comment]
This is now using the latest version of `Makefile.common` that deals with the weirdness we were seeing with `staticcheck` and Go modules. It also enables additional checks that have been fixed along the way.

**Comment by matthiasr** [comment]
Wonderful, thank you!

**Pull request merged**